languages/swift/exercises/concept/closures/.docs/instructions.md

```diff
@@ -0,0 +1,111 @@
+In this exercise you will be using closures with a higher-order function to generate sequences of numbers from a seed number.
+
+You are given the function
+
+```swift
+unroll(_: Int,
+       generator: (Int) -> (seed: Int, value: Int),
+       stop: (Int) -> Bool
+      ) -> [Int]
+```
+
+`unroll` takes three parameters, a seed value; a `generator` function which, when applied to the seed returns a tuple consisting of the next value for the seed and the next value for the sequence; and a `stop` function which, when applied to the current seed value returns true if the sequence generation is finished. Unroll then returns the generated sequence. For example, to generate the powers of 3 up to 1,000,000, we can write:
+
+```swift
+let powersOf3 = unroll(1,
+                       generator: { (n: Int) -> (seed: Int, next: Int) in (seed: 3 * n, next: n) },
+                       stop: { (n: Int) -> Bool in n > 1_000_000 })
+// => [1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147, 531441]
+```
+
+You will use unroll to help you generate a few sequences.
+
+## 1. Write helper closures to generate a sequence of digits
+
+One can generate a sequence of digits of a number, _n_ by computing `n % 10` to get the last digit of `n` and then dividing `n` by 10 to remove that last digit. This process can then be repeated until `n` is 0.
+
+For this task, you need to define a closure of type `(Int) -> (seed: Int, next: Int)` That takes a number and returns that number with its last digit removed as `seed:` along with the last digit as `next:`. Assign this closure to a constant named `generateDigit`.
+
+You will also need to generate a closure of type `(Int) -> Bool` That returns `true` if n = 0 and `false` otherwise. Assign this closure to a constant named `stopAtZero`.
```

Comment:
```suggestion
You will also need to generate a closure of type `(Int) -> Bool` that returns `true` if n = 0 and `false` otherwise. Assign this closure to a constant named `stopAtZero`.
```